stdlib/multiprocessing/sharedctypes.pyi

```diff
@@ -1,43 +1,122 @@
+    *args: Any,
+    lock: Union[Literal[True], _LockLike],
+    ctx: Optional[BaseContext] = ...,
+) -> SynchronizedBase[Any]: ...
+@overload
+def Value(
+    typecode_or_type: Union[str, Type[_CData]], *args: Any, lock: Union[bool, _LockLike] = ..., ctx: Optional[BaseContext] = ...
+) -> Any: ...
+@overload
+def Array(
+    typecode_or_type: Type[_CT],
+    size_or_initializer: Union[int, Sequence[Any]],
+    *,
+    lock: Literal[False],
+    ctx: Optional[BaseContext] = ...,
+) -> _CT: ...
+@overload
+def Array(
+    typecode_or_type: Type[_CT],
+    size_or_initializer: Union[int, Sequence[Any]],
+    *,
+    lock: Union[Literal[True], _LockLike],
+    ctx: Optional[BaseContext] = ...,
+) -> SynchronizedArray[_CT]: ...
+@overload
+def Array(
+    typecode_or_type: Union[str, Type[_CData]],
+    size_or_initializer: Union[int, Sequence[Any]],
+    *,
+    lock: Union[Literal[True], _LockLike],
+    ctx: Optional[BaseContext] = ...,
+) -> SynchronizedArray[Any]: ...
+@overload
 def Array(
     typecode_or_type: Union[str, Type[_CData]],
     size_or_initializer: Union[int, Sequence[Any]],
     *,
     lock: Union[bool, _LockLike] = ...,
     ctx: Optional[BaseContext] = ...,
-) -> _Array: ...
-def Value(
-    typecode_or_type: Union[str, Type[_CData]], *args: Any, lock: Union[bool, _LockLike] = ..., ctx: Optional[BaseContext] = ...
-) -> _Value: ...
+) -> Any: ...
+def copy(obj: _CT) -> _CT: ...
+@overload
+def synchronized(obj: _SimpleCData[_T], lock: Optional[_LockLike] = ..., ctx: Optional[Any] = ...) -> Synchronized[_T]: ...
+@overload
+def synchronized(obj: ctypes.Array[c_char], lock: Optional[_LockLike] = ..., ctx: Optional[Any] = ...) -> SynchronizedString: ...
+@overload
+def synchronized(obj: ctypes.Array[_CT], lock: Optional[_LockLike] = ..., ctx: Optional[Any] = ...) -> SynchronizedArray[_CT]: ...
+@overload
+def synchronized(obj: _CT, lock: Optional[_LockLike] = ..., ctx: Optional[Any] = ...) -> SynchronizedBase[_CT]: ...
+
+class _AcquireFunc(Protocol):
+    def __call__(self, block: bool = ..., timeout: Optional[float] = ...) -> bool: ...
+
+class SynchronizedBase(ContextManager[bool], Generic[_CT]):
+    acquire: _AcquireFunc = ...
+    release: Callable[[], None] = ...
+    def __init__(self, obj: Any, lock: Optional[_LockLike] = ..., ctx: Optional[Any] = ...) -> None: ...
+    def __reduce__(self) -> Tuple[Callable[..., Any], Tuple[Any, _LockLike]]: ...
+    def get_obj(self) -> _CT: ...
+    def get_lock(self) -> _LockLike: ...
+
+class Synchronized(SynchronizedBase[_SimpleCData[_T]], Generic[_T]):
+    value: _T = ...
+
+class SynchronizedArray(SynchronizedBase[ctypes.Array[_CT]], Generic[_CT]):
+    def __len__(self) -> int: ...
+    @overload
+    def __getitem__(self, i: int) -> _CT: ...
+    @overload
+    def __getitem__(self, s: slice) -> List[_CT]: ...
+    @overload
+    def __setitem__(self, i: int, o: _CT) -> None: ...
+    @overload
+    def __setitem__(self, s: slice, o: Iterable[_CT]) -> None: ...
```

Comment:
At least on Python 3.7 and 3.9, `SynchronizedArray` only allows the non-slice form.